web/src/main/java/org/wso2/testgrid/web/operation/PipelineManager.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.testgrid.web.operation;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.hc.client5.http.fluent.Request;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.HttpResponse;
+import org.apache.hc.core5.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.wso2.testgrid.web.utils.Constants.BLUE_OCEAN_URI;
+import static org.wso2.testgrid.web.utils.Constants.JENKINS_HOME;
+import static org.wso2.testgrid.web.utils.Constants.JENKINS_USER_AUTH_KEY;
+
+import java.io.IOException;
+
+import static javax.ws.rs.core.HttpHeaders.USER_AGENT;
+
+/**
+ * Implementation of creating pipeline jobs from config.xml files.
+ */
+public class PipelineManager {
```

Comment:
Your javadoc says that this is 'creating' pipelines. So, can we use the same for ur class name as well? :)
Something like `JenkinsPipelineCreator`.

You can actually drop 'Jenkins' prefix from all these class names if u have a sub-package named 'jenkins'.

Reply:
I think we will need to handle job deletion and may be job change functions in future, Therefore I think changing this to **JenkinsPipelineManager** and change the javadoc according would be better. WDYT?